### PR TITLE
NODE-1832: Peer connections on startup

### PR DESCRIPTION
**Pull request by chepiov**

### CODE REVIEW

File: node/src/main/scala/com/wavesplatform/network/PeerDatabase.scala [hunk 1]
```diff
@@ -49,6 +51,8 @@ object PeerDatabase extends ScorexLogging {
 
     override def randomPeer(excluded: Set[InetSocketAddress]): Option[InetSocketAddress] = None
```
Comment (Karasiq) [inline node/src/main/scala/com/wavesplatform/network/PeerDatabase.scala]: Better remove `randomPeer` method completely and use the `randomPeers(1, ...)` as a replacement

Comment (chepiov) [reply node/src/main/scala/com/wavesplatform/network/PeerDatabase.scala]: Done

File: node/src/main/scala/com/wavesplatform/network/NetworkServer.scala [hunk 2]
```diff
@@ -202,37 +206,47 @@ object NetworkServer extends ScorexLogging {
         }
       )
 
-    val connectTask = workerGroup.scheduleWithFixedDelay(1.second, 5.seconds) {
-      import scala.collection.JavaConverters._
-      val outgoing = outgoingChannels.keySet.iterator().asScala.toVector
+    def scheduleConnectTask(): Unit = {
```
Comment (phearnot) [inline node/src/main/scala/com/wavesplatform/network/NetworkServer.scala]: Please avoid `return` at all costs!
```suggestion
    def scheduleConnectTask(): Unit = if (!shutdownInitiated) {
```

File: node/src/main/scala/com/wavesplatform/network/HandshakeHandler.scala [hunk 3]
```diff
@@ -31,12 +31,12 @@ class HandshakeDecoder(peerDatabase: PeerDatabase) extends ReplayingDecoder[Void
 
 case object HandshakeTimeoutExpired
 
-class HandshakeTimeoutHandler(handshakeTimeout: FiniteDuration) extends ChannelInboundHandlerAdapter with ScorexLogging {
+class HandshakeTimeoutHandler(handshakeTimeout: => FiniteDuration) extends ChannelInboundHandlerAdapter with ScorexLogging {
```
Comment (Karasiq) [inline node/src/main/scala/com/wavesplatform/network/HandshakeHandler.scala]: Better do `handshakeTimeout` an explicit function 
```suggestion
class HandshakeTimeoutHandler(handshakeTimeout: () => FiniteDuration) extends ChannelInboundHandlerAdapter with ScorexLogging {
```

Comment (phearnot) [reply node/src/main/scala/com/wavesplatform/network/HandshakeHandler.scala]: Actually, looks like there's no need to use a by-name parameter at all here.

**Pull request merged**